test/jdk/javax/swing/JFileChooser/FileSystemView/SystemIconTest.java

```diff
@@ -73,30 +73,39 @@ static void negativeTests() {
     static void testSystemIcon(File file, boolean implComplete) {
         int[] sizes = new int[] {16, 32, 48, 64, 128};
         for (int size : sizes) {
-            ImageIcon icon = (ImageIcon) fsv.getSystemIcon(file, size, size);
+            Icon i = fsv.getSystemIcon(file, size, size);
 
+            if (i == null) {
+                throw new RuntimeException(file.getAbsolutePath() + " icon is null");
+            }
+
+            if (!(i instanceof ImageIcon)) {
+                // Default UI resource icon returned - it is not covered
+                // by new implementation so we can not test it
+                continue;
```

Comment:
I see that we are hitting this condition for all sizes on my 10.15.7 mac. Basically making testSystemIcon() to not run in my Mac. Do we have any use cases where we dont hit this condition in Mac or any other non-Windows platform?

Reply:
I tested on Ubuntu - it does not hit that condition. In future, when similar functionality with multiresolution icons might be implemented on Mac we might pass the test on Mac in similar way but right now it is implemented using the UIResourceIcon and that is - while being within spec - makes test to fail.